### ISSUE TITLE
Title alignment workarounds, and table fixes.

### ISSUE THREAD

**Issue by erdnuesse** [wlcsm/resume#4]
2 Days, And I must say, there is some nasty workaround with 
`\# This space ensures the left-aligned is at the same level as the following`
`\# right-aligned text.`

one one liner inbetween, and the next entry below it blows up and seems unfixable. 
all subsequent ones again, seem fine.

Some entries were better with tables, I must say, but that one valign bug I cannot seem to get around. 

just  try and insert one line of:

`.item "Sabbatical" " " "Jan - Oct 1900"
`
into the work experience. 

I could not yet find a way to realign. 
Currently, it's 4 am, and I think about replacing all .items wit one-row tables, similar to 
```
.de mytable
.LP
.in -.9i
.TBL border= cols=2 width='75% 25%' cpd=0 tal=c
.TR 
.TD bgc= hal=l \m[grey]\fB\s(12\\$1\fR\m[]
.TD bgc= hal=r \m[grey]\fB\s(11\\$2\fR\m[]
.ETB
.sp -.7
.in
..
```

(yes, I use hdtbl)
I now have changed basically every item rearrangement with a table, sometimes just one-row-tables, other times multiline cells, because table paddings are hell to workaround as a newbie, so yea, I'm pretty ashamed as to how my code looks, but the layouting is way easier to work with - but not at all reusable. Many copy pastas and spaghetti.

**Comment by wlcsm** [comment]
Hey @erdnuesse,

Thanks for reporting an issue!

I'm not 100% sure under what conditions this bug is occurring. Could you provide a minimum example so that I can better understand and reproduce the issue?

**Comment by erdnuesse** [comment]
Sure.

```
\# Will Resume 2023
.
.ds EMAIL wcashman1351@gmail.com
.ds PHONE +44 7818114085
.ds GITHUB wlcsm
.ds LINKEDIN cashman-will
.
\# Define the page header as empty, thus removing page numbers.
.ds CH
.
\# Helvetica font
.fam H
.
\# Margin. Create a right margin of 0.5i by subtracting the left margin
\# plus 0.5i from the width of the entire page.
.nr PO 0.7i
.nr HM 1.0i
.nr FM 0.5i
.nr LL 21c
.nr LL -\n[PO]
.nr LL -0.5i
.
\# Code block
.de code_begin
.fam C
\\$*
..
.
.de code_end
.fam
..
.
\# Hyperlinks
\# 1. Display name
\# 2. Hyperlink destination
.de url
.pdfhref W -D \\$2 \m[grey]\\$1\m[black]
..
.
.de code_url
.code_begin
.sp -0.8
.url \\$1 \\$2
.code_end
..
.
\# Import the template macros
.so resume.tmac
.
\# Main document starts here
.title Will Cashman
.
\s(12\m[grey]Full Stack Go/Python Engineer
.
.sp -4
.DS R
.nf
\m[grey]
Oxford, UK
.url "\*[EMAIL]" mail:\*[EMAIL]
\m[grey]
\*[PHONE]
.url "Github: wlcsm" "https://github.com/wlcsm"
.url "LinkedIn: \*[LINKEDIN]" "https://www.linkedin.com/in/\*[LINKEDIN]/"
.DE
.
.ps 12
.sp 1.5
.
.
.section WORK EXPERIENCE
.
.
.item "CrowdStrike" "Software Engineer, Remote" "Dec 2022 - Sep 2023"
.list Developed internal tooling and microservices in Python and Golang on AWS and Kubernetes to increase developer productivity and reliability of the LogScale product.
.list Coordinated upgrades of 80,000+ fleet of Ubuntu servers using Python and Ansible, in addition to general Linux administration.
.list Spearhead implementation and deployment of better targeted testing procedures in CI/CD.
.list Routinely resolved performance bottlenecks to improve efficiency and reduce infrastructure costs.
.
.item "Sabbatical" " " "Jan - Oct 1900"
.
.item "TikTok" "Software Engineer, Shenzhen, CN" "Aug 2021 - Aug 2022"
.list Follow an agile release train to design and develop core CI/CD capabilities for company release platform by building Go and Python microservices and tooling.
.list Lead the sucessful integration of a competing internal product. Involving a complete system migration and the development of a bespoke cross platform data migration tool.
.list Actively engaged with users to seek feedback and resolve obstacles, resulting in a 85% user retention rate over a couple of months of release.
.list Used Python to perform system migrations, automate tasks, and SQL database maintenance.
.
```

tl;dr: 
as soon as there's no "list" under the "item" all falls apart for that line. 
Everything seems too interdependent from each other. That's why I refused to have recursive formatting in my version, and inserted the 1-row table with 2 cells, to mitigate the up- and down shifting of text segments.

**Comment by wlcsm** [comment]
Thanks for explaining! I understand the problem now.

I tried for a unreasonable amount of time to get it to work with `.mk` and `.rt`, but it simply refused to work as expected, which is probably why I put that hacky solution in there in the first place.

In the end I've settled on yet another hacky solution: manually shifting the right-aligned text upwards.
If you have the time, please check out the PR and verify the empty item's text is aligned for your resume.

Your solution is really nice and clean, but I think I'd like to keep everything inside vanilla groff + ms macros while I can. When I get some more free time I'd love to check out hdtbl as it looks really useful.

**Comment by erdnuesse** [comment]
First of all, I don't think my approach is clean. However, as I am not an expert in groff, the limitation of not being able to have two different positioning types left and right-bound in one line made a workaround necessary. Having two formatable cells instead of two lines that merge into one with positioning is probably the cleaner one. 

One has to identify the priorities, a) one more dependency vs. b) probably quick and dirty fixes. 

One more thing: as I have found no reliable way to autosize cell widths, in terms of the 100% body, / left / right columns, I needed more than one table style to accomodate left/right width combinations. ie. 90% left 10% right and 85/15, just because my needs were different in different sections due to certificate names / date widths. 
That means ymmv between one and potentially multiple table predefinitions to choose from. Best case, one size fits all, worst case many different sizes for any time you wish to make use of this formatting style. 

In the end, I am not sure, if TEX may have a better solution overall. But that's almost 2GB of dependency hell that I am not sure I want to go back to - as I came from LATEX / moderncv and wanted to try something more lean.

**Comment by wlcsm** [comment]
That makes sense. Sounds like the table solution suits your needs better.

> In the end, I am not sure, if TEX may have a better solution overall. But that's almost 2GB of dependency hell that I am not sure I want to go back to - as I came from LATEX / moderncv and wanted to try something more lean.

I think we all are coming to groff to escape Tex 😄
